Index: gnuradio-core/src/lib/general/gr_head.h
===================================================================
--- gnuradio-core/src/lib/general/gr_head.h	(revision 8484)
+++ gnuradio-core/src/lib/general/gr_head.h	(working copy)
@@ -42,6 +42,7 @@
   int	d_ncopied_items;
 
  public:
+  void rewind() { d_ncopied_items=0; }
   int work (int noutput_items,
 		 gr_vector_const_void_star &input_items,
 		 gr_vector_void_star &output_items);
Index: gnuradio-core/src/lib/general/gr_head.i
===================================================================
--- gnuradio-core/src/lib/general/gr_head.i	(revision 8484)
+++ gnuradio-core/src/lib/general/gr_head.i	(working copy)
@@ -24,6 +24,8 @@
 class gr_head : public gr_block {
   friend gr_block_sptr gr_make_head (size_t sizeof_stream_item, int nitems);
   gr_head (size_t sizeof_stream_item, int nitems);
+  public:
+    void rewind() { d_ncopied_items=0; }
 };
 
 %rename(head) gr_make_head;
